src/main/resources/api/paths/calendar/calendars/calendarIds.yaml

```diff
@@ -0,0 +1,101 @@
+get:
+  summary: Get calendar(s)
+  description:
+    Get one or more calendar's information by their ID.  All calendars must exist for a successful
+    response, otherwise a 404 will be returned.
+  operationId: getCalendars
+  parameters:
+    - in: path
+      name: calendarIds
+      required: true
+      style: simple
+      explode: true
+      schema:
+        type: array
+        items:
+          type: string
+          format: uuid
+        minItems: 1
+      description:
+        A list of calendar IDs to operate on, separated by commas.  Duplicate IDs will be ignored.
+  responses:
+    "200":
+      description: The query results
+      content:
+        application/json:
+          schema:
+            $ref: ../../../schemas/calendarCollection.yaml
+    "400":
+      $ref: ../../../errors/invalidRequest.yaml
+    "404":
+      $ref: ../../../errors/calendarNotFound.yaml
+    "500":
+      $ref: ../../../errors/internalServerError.yaml
+put:
+  summary: Update an existing calendar
+  description:
+    Overwrite an existing calendar with the provided payload.  The provided calendar must already
+    exist (attempting to overwrite a calendar that does not yet exist will result in a 404).  If the
+    payload includes any IDs, they will be ignored, and the existing calendar ID reused.
+  operationId: updateCalendar
+  parameters:
+    - in: path
+      # should not be plural, but must be consistent with the overall path variable.
+      # api-lint gets mad if we declare two separate paths, /{calendarId} and /{calendarIds},
+      # so we are stuck naming this variable "calendarIds" despite it being only a single id
```

Comment:
An alternative design that would avoid this semantic conflict between paths that should not be the same could be to use query string parameters for the endpoint that fetches multiple calendars and reserve the single record path for operations related to a single record.

Reply:
Personally, I felt the mismatching plural just in this source code file would be preferred to making semantically different endpoints (with query parameters, other paths, etc.).  The extra `s` is only present in this file; in the code it's and generated documentation it's `calendarId` as expected.

Reply:
> Personally, I felt the mismatching plural just in this source code file would be preferred to making semantically different endpoints (with query parameters, other paths, etc.). 

I disagree with this because the endpoints are semantically different. One describes access to a single record and the other describes access to a filtered collection. I think it is confusing that one is masquerading as the other.

To me, it is conventional in HTTP APIs that multiple value parameters tend to be represented either in the body or query string parameters rather than part of the path directly. A path with commas in seems rather odd to me.

> The extra `s` is only present in this file; in the code it's and generated documentation it's `calendarId` as expected.

I chose to put the comment here because this comment made it obvious the developers knew about this design trade off. My concerns are not with how this manifests in the internal code, rather how it manifests in the API from the outside.

Reply:
There's not much for the GET endpoint that would be different from adding them to the collection search endpoint, however, if we were to do that I'm not sure what to do with the multi-DELETE endpoint.  Imo, moving the multi-GET endpoint without the DELETE one just makes the issue stronger.

If you have a suggestion of how better to present the DELETE endpoint, I am all ears.  Something like `calendars/delete?id=aaaaa&id=bbbbb...` or `calendars/delete?id=aaaaa,bbbbb,ccccc...` could work, but then that seems like a bit of a conflict with the PUT endpoint `calendars/aaaaa` (same with sending a request body to the delete endpoint).  I think consolidating the multi-GET endpoint to the search endpoint would work, but I don't think it would provide any additional clarity without a solution for the multi-DELETE endpoint, too.

The comma-separated lists are a bit funny looking, but I have seen it around and it's [the default for Swagger/OAS path parameters](https://swagger.io/docs/specification/serialization/).  Also, for what it's worth, I think it looks a little more RESTful, with the path reflecting the entities rather than the method.

Reply:
> There's not much for the GET endpoint that would be different from adding them to the collection search endpoint, however, if we were to do that I'm not sure what to do with the multi-DELETE endpoint. Imo, moving the multi-GET endpoint without the DELETE one just makes the issue stronger.

Personally, I would move the DELETE endpoint to use a query parameter as well.

To me, query parameters are a more expressive way of representing these kinds of parameters than the path (which I prefer to keep simple).

> If you have a suggestion of how better to present the DELETE endpoint, I am all ears. Something like `calendars/delete?id=aaaaa&id=bbbbb...` or `calendars/delete?id=aaaaa,bbbbb,ccccc...` could work

Something like that would be my preference.

> but then that seems like a bit of a conflict with the PUT endpoint `calendars/aaaaa` (same with sending a request body to the delete endpoint). 

How is that in conflict with the PUT endpoint?

To me, the PUT endpoint is specifically about a single record. I think it's a good thing that takes a different form to endpoints that interact with multiple records.

(I'm kinda aware that even talking about records and entities is somewhat of a trap from a ReSTful perspective)

> The comma-separated lists are a bit funny looking, but I have seen it around and it's [the default for Swagger/OAS path parameters](https://swagger.io/docs/specification/serialization/). 

Maybe they know more than I do. I find it an unusual form.

You mentioned an example, what would that example be?

> Also, for what it's worth, I think it looks a little more RESTful, with the path reflecting the entities rather than the method.

What do you mean by method? (to me, the method in HTTP refers to GET, POST etc)

Reply:
Ultimately, this is largely stylistic feedback and is heavily subjective (much like many design choices) and should be taken that way.

Reply:
> Thoughts on these? 

I think you reasoned this through for yourself.

I wasn't thinking of using the keyword form you described. 

Rather that operations involving multiple records would be at the collection path and operations involving a single specific record at the record path, with differentiation by method and supported parameters.

FOLIO tends to distinguish between single record and collection paths.

This is nearly what you have in the last example. I might tweak it a little:
```
// Collection paths for operations on the collection
GET /calendars?ids=A,B (I'm ambivalent about this form vs. repetitions of the parameter id=A&id=B)
DELETE /calendars?ids=B,C

// individual record path for operations on the individual record
PUT /calendars/{id}
```

Reply:
That sounds good.  For consistency, do you think it would be a good idea to also support plain GET/DELETE of individual record paths, too?  Or leave it to just the collection endpoints

Reply:
> For consistency, do you think it would be a good idea to also support plain GET/DELETE of individual record paths, too? Or leave it to just the collection endpoints

That's a trade off I'm not sure how best to advise you on. 

Each endpoint requires maintenance and increases the surface area of the API.

If there isn't a use for those endpoints now, I wouldn't implement them. 

There are defacto standard FOLIO API structures. However this interface already moves away from them in a few ways: 
* using specific parameters for searching over CQL
* not using the standard change metadata
* not using the standard error reporting

Given those variations, there is little value in trying to be consistent with the endpoint paths.

Reply:
Done.   You can see the specific changes [here](https://github.com/folio-org/mod-calendar/commit/a7c6088cd15c7dac1ba1afe789983edc851a85ce) and the updated API docs [here](https://ualibweb.github.io/ua-folio-docs/calendar.html).